### DISCUSSION TITLE
Interaction with fzf-tab in zsh

### DISCUSSION THREAD

**Discussion by norman-zon** [carapace-sh/carapace-bin#2596] [Q&A]
I wonder about the interaction of carapace with completions from fzf-tab.

To be fair, I am very new to carapace, but I believe I read through all docs and issues regarding carapace and fzf-tab.

I have fzf-tab setup like this:
```
# enable searching of display strings for everything
zstyle ':completion:*' fzf-search-display true
# bind TAB to toggle select and move down (default is just move, toggle is CTRL+SPACE)
zstyle ':fzf-tab:complete:*' fzf-bindings 'space:toggle+down'
# preview directory's content with exa when completing cd
zstyle ':fzf-tab:complete:cd:*' fzf-preview 'eza -1 --color=always $realpath'
# prefix query to search
zstyle ':fzf-tab:*' query-string prefix first
```

and carapace is setup like this:

```
export CARAPACE_BRIDGES='zsh,fish,bash,inshellisense' # optional
zstyle ':completion:*' format $'\e[2;37mCompleting %d\e[m'
source <(carapace _carapace)
```

The completion for `git` looks like this:

![image](https://github.com/user-attachments/assets/78c642e2-3dd9-47a0-94d8-a55d872fba65)

I don't understand why all completion groups are shown at once. Can someone shed some light into this?

**Comment by rsteube** [comment]
This should be related to the `format` zstyle recommended by carapace:
```zsh
zstyle ':completion:*' format $'\e[2;37mCompleting %d\e[m'
```
I've seen the same behaviour with the official git completion when it is set.

Might be that the pattern needs to be more specific. Or related to a different config.
Haven't delved that deeply into this issue yet.

See _Formatting The Display_ and _Grouping Results_ in https://thevaluable.dev/zsh-completion-guide-examples/ .
Maybe there's something useful.

**Reply by rsteube** [reply]
Does this help?
```sh
zstyle ':completion:*' group-name ''
```

**Reply by norman-zon** [reply]
> Does this help?
> 
> ```shell
> zstyle ':completion:*' group-name ''
> ```

This does not change the behaviour. 
>See Formatting The Display and Grouping Results in https://thevaluable.dev/zsh-completion-guide-examples/ .
Maybe there's something useful.

Thanks for linking the article above, I just skimmed it very quickly, but it looks very interesting. Will dive into it later.
